repo/content/index/one_use_index_builder.go

```diff
@@ -0,0 +1,157 @@
+package index
+
+import (
+	"crypto/rand"
+	"hash/fnv"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/petar/GoLLRB/llrb"
+
+	"github.com/kopia/kopia/internal/gather"
+)
+
+// Less compares with another *Info by their ContentID and return true if the current one is smaller.
+func (i *Info) Less(other llrb.Item) bool {
+	return i.ContentID.less(other.(*Info).ContentID) //nolint:forcetypeassert
+}
+
+// OneUseBuilder prepares and writes content index for epoch index compaction.
+type OneUseBuilder struct {
+	indexStore *llrb.LLRB
+}
+
+// NewOneUseBuilder create a new instance of OneUseBuilder.
+func NewOneUseBuilder() *OneUseBuilder {
+	return &OneUseBuilder{
+		indexStore: llrb.New(),
+	}
+}
+
+// Add adds a new entry to the builder or conditionally replaces it if the timestamp is greater.
+func (b *OneUseBuilder) Add(i Info) {
+	cid := i.ContentID
+
+	found := b.indexStore.Get(&Info{ContentID: cid})
+	if found == nil || contentInfoGreaterThanStruct(&i, found.(*Info)) { //nolint:forcetypeassert
+		_ = b.indexStore.ReplaceOrInsert(&Info{
```

Comment:
wouldn't `&i` be sufficiient? The structure is passed by value.

Reply:
Yes, it seems this should have been modified together when removing `InfoCompact`.
On the other hand, this should make no difference to memory usage since the unnecessary variables are local variables on stack.

I submitted a PR #4167 for it, feel free to merge or close it.